subprojects/platform-play/src/main/java/org/gradle/play/internal/twirl/TwirlCompilerAdapterV13X.java

```diff
@@ -0,0 +1,94 @@
+ */
+
+package org.gradle.play.internal.twirl;
+
+import org.gradle.language.twirl.TwirlImports;
+import org.gradle.language.twirl.TwirlTemplateFormat;
+import org.gradle.scala.internal.reflect.ScalaCodecMapper;
+import org.gradle.scala.internal.reflect.ScalaMethod;
+import org.gradle.scala.internal.reflect.ScalaReflectionUtil;
+
+import java.io.File;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+class TwirlCompilerAdapterV13X extends TwirlCompilerAdapterV10X {
+    private static final Iterable<String> SHARED_PACKAGES = Arrays.asList("play.twirl.compiler", "scala.io", "scala.util.parsing.input", "scala.collection");
+
+    public TwirlCompilerAdapterV13X(String twirlVersion, String scalaVersion) {
+        super(twirlVersion, scalaVersion);
+    }
+
+    @Override
+    public ScalaMethod getCompileMethod(ClassLoader cl) throws ClassNotFoundException {
+        return ScalaReflectionUtil.scalaMethod(
+            cl,
+            "play.twirl.compiler.TwirlCompiler",
+            "compile",
+            File.class,
+            File.class,
+            File.class,
+            String.class,
+            cl.loadClass("scala.collection.Seq"),
+            cl.loadClass("scala.collection.Seq"),
+            cl.loadClass(ScalaCodecMapper.getClassName()),
+            boolean.class
+        );
+    }
+
+    @Override
+    public Object[] createCompileParameters(ClassLoader cl, final File file, File sourceDirectory, File destinationDirectory, TwirlImports defaultImports, TwirlTemplateFormat templateFormat, List<String> additionalImports) throws ClassNotFoundException, NoSuchMethodException, InvocationTargetException, InstantiationException, IllegalAccessException {
+        return new Object[]{
+            file,
+            sourceDirectory,
+            destinationDirectory,
+            templateFormat.getFormatType(),
+            toScalaSeq(getDefaultImports(defaultImports), cl),
+            toScalaSeq(Collections.emptyList(), cl),
+            ScalaCodecMapper.create(cl, "UTF-8"),
+            isInclusiveDots(),
+        };
+    }
+
+    private Object toScalaSeq(Collection<?> list, ClassLoader classLoader) {
+        ScalaMethod method = ScalaReflectionUtil.scalaMethod(classLoader, "scala.collection.JavaConversions", "asScalaBuffer", List.class);
+        Object scalaBuffer = method.invoke(list);
+        return scalaBuffer;
+    }
+
+    private boolean isInclusiveDots() {
+        return false;
+    }
+
+    @Override
+    public Iterable<String> getClassLoaderPackages() {
+        return SHARED_PACKAGES;
+    }
+
+    @Override
+    public String[] getDependencyNotation() {
+        if (scalaVersion.startsWith("2.12")) {
+            // We need scala.util.parsing.input.Positional
+            return new String[]{super.getDependencyNotation()[0], "org.scala-lang.modules:scala-parser-combinators_2.12:1.0.6"};
```

Comment:
This looks a bit fragile to me. I mean, if `super` starts to return an array with more than a single element, this will stop working. So, I would so something like:

```java
String[] dependencyNotation = super.getDepenencyNotation();
String[] result = Arrays.copyOf(dependencyNotation, dependencyNotation.length +1);
result[dependencyNotation.length] = "org.scala-lang.modules:scala-parser-combinators_2.12:1.0.6";
return result;
```

Reply:
Maybe we can later refactor `getDependencyNotation` to return a list instead (easier to manipulate)  since this is an internal API.

Reply:
Thanks, let me make it an array.

Reply:
Sorry, I mean to make it a list.